<a href="https://colab.research.google.com/github/KabitaPaudel675/Kabita_AI_Assignments/blob/main/KP_AI_Assignment_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

This dataset consists of corn yield of 12 states from Southeastern United States(Alabama,Arkansas, Florida, Georgia, Kentucky, Lousiana, Mississipi, North Carolina, South Carolina, Tennessee, Virginia, West Virginia) in the year 1980 and 2020. In this exercise we are comparing mean corn yield of 1980 with corn yield of 2020 in the southeast US. 



In [ ]:
## Open datafile and store into Pandas dataframe 
df=pd.read_csv('https://raw.githubusercontent.com/KabitaPaudel675/AI_Assignment-1/main/KY_CY_1980_2020.csv?token=GHSAT0AAAAAABYQV5YXD6MKNBT5N5C3OL5GYZ3CJNA')
df

Year     1980      2018
0          Alabama  2510.80  10859.20
1         Arkansas  2196.95  11424.10
2          Florida  2824.65   9854.89
3          Georgia  2510.80  10670.90
4         Kentucky  4268.36  11110.30
5         Lousiana  3075.73  10670.90
6       Missisippi  2259.72  11612.50
7   North Carolina  3766.20   7657.94
8   South Carolina  2950.19   7783.48
9        Tennessee  3452.35  10922.00
10        Virginia  3766.20   9289.96
11   West Virginia  4896.06   9541.04

In [ ]:
df.describe()

1980          2018
count    12.000000     12.000000
mean   3206.500833  10116.434167
std     841.066187   1325.309102
min    2196.950000   7657.940000
25%    2510.800000   9478.270000
50%    3012.960000  10670.900000
75%    3766.200000  10969.075000
max    4896.060000  11612.500000

In [ ]:
df.shape ## No of rows and columns of new dataframe

(12, 3)

In [ ]:
df.isnull() ## Checking if we have null 


Year   1980   2018
0   False  False  False
1   False  False  False
2   False  False  False
3   False  False  False
4   False  False  False
5   False  False  False
6   False  False  False
7   False  False  False
8   False  False  False
9   False  False  False
10  False  False  False
11  False  False  False

For this exercise, we are doing paired t test comparing means of corn yield of Southeastern US in 1980 and 2020. Since we are comparing means of the same area in two different years, we are using paired t test for depoendent samples.
The  t-test works by checking the means from two samples to see if they are significantly different from each other. It does this by calculating the standard error in the difference between means, which can be interpreted to see how likely the difference is, if the two samples have the same mean (the null hypothesis).


In [ ]:

# Paired t-test for dependent samples

from scipy.stats import ttest_rel

In [ ]:
data1=df['1980']

In [ ]:
data2=df['2018']

In [ ]:
# compare samples
stat, p = ttest_rel(data1, data2)
print('Statistics=%.3f, p=%.3f' % (stat, p))

Statistics=-13.187, p=0.000


The test result shows the test statistic "t" is "-13.187" with p-value 0.000 that means the corn yield has decreased from 1980 to 2020 in the 12 Southeast US states and the decrement in the yield is highly statistically significant since p-value is less than 0.05.